hilti/runtime/CMakeLists.txt

```diff
@@ -74,19 +76,58 @@ foreach ( lib hilti-rt hilti-rt-debug )
     # jrx-objects which is an OBJECT library itself and depended on by
     # `jrx-objects`.
     target_link_libraries(${lib} jrx-objects)
-endforeach ()
 
-# Build hilti-rt with release flags.
-string(REPLACE " " ";" cxx_flags_release ${CMAKE_CXX_FLAGS_RELEASE})
-target_compile_options(hilti-rt-objects     PRIVATE ${cxx_flags_release})
-target_compile_options(hilti-rt-objects     PRIVATE "-g;-O3;-DNDEBUG;-Wall")
-target_compile_definitions(hilti-rt-objects PRIVATE "HILTI_RT_BUILD_TYPE_RELEASE")
-
-# Build hilti-rt-debug with debug flags.
-string(REPLACE " " ";" cxx_flags_debug ${CMAKE_CXX_FLAGS_DEBUG})
-target_compile_options(hilti-rt-debug-objects     PRIVATE ${cxx_flags_debug})
-target_compile_options(hilti-rt-debug-objects     PRIVATE "-UNDEBUG;-O0;-Wall")
-target_compile_definitions(hilti-rt-debug-objects PRIVATE "HILTI_RT_BUILD_TYPE_DEBUG")
+    if ("${lib}" MATCHES "debug$")
+        string(REPLACE " " ";" cxx_flags_debug ${CMAKE_CXX_FLAGS_DEBUG})
+        target_compile_options(hilti-rt-debug-objects     PRIVATE ${cxx_flags_debug})
+        target_compile_options(hilti-rt-debug-objects     PRIVATE "-UNDEBUG;-O0;-Wall")
+        target_compile_definitions(hilti-rt-debug-objects PRIVATE "HILTI_RT_BUILD_TYPE_DEBUG")
+    else ()
+        string(REPLACE " " ";" cxx_flags_release ${CMAKE_CXX_FLAGS_RELEASE})
+        target_compile_options(hilti-rt-objects     PRIVATE ${cxx_flags_release})
+        target_compile_options(hilti-rt-objects     PRIVATE "-g;-O3;-DNDEBUG;-Wall")
+        target_compile_definitions(hilti-rt-objects PRIVATE "HILTI_RT_BUILD_TYPE_RELEASE")
+    endif ()
+
+    if (${HILTI_DEV_PRECOMPILE_HEADERS})
+        # Precompile libhilti for use in JIT during development. Users should use
+        # `scripts/precompile-headers.sh` instead.
+        #
+        # We generate the precompiled header by hand and only use it during JIT,
+        # but e.g., not to during compilation of Spicy itself. This gives us the
+        # benefits of JIT without e.g., making it harder for ccache to work during
+        # development. It also allows us to punt on some trickier cleanups of
+        # header files.
+        set(debug_opt "")
+        if ("${lib}" MATCHES "debug$")
+            set(debug_opt "_debug")
+        endif ()
+
+        # NOTE: The compiler invocation here should be kept in sync
+        # with what we do in `scripts/precompile-headers.sh`.
+        get_target_property(PCH_INCLUDES ${lib}-objects INCLUDE_DIRECTORIES)
+        list(TRANSFORM PCH_INCLUDES PREPEND "-I")
+        get_target_property(PCH_OPTIONS ${lib}-objects COMPILE_OPTIONS)
+        add_custom_command(
```

Comment:
The shell script now gained `--hilti-config` but CMake still isn't calling this. That on purpose? If it's too difficult to have CMake use the script instead, then that's fine; but if it's doable, I'd still prefer it.

Reply:
I forgot to implement this after you suggested it above. The script is now also used from the CMake setup.